[Automation] javac compile fails for org.openjfx:javafx-base:13-ea+14

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="13-ea+14" ./gradlew clean compileTestJava -Pcoordinates="org.openjfx:javafx-base:11.0.2"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/24753645127/job/72422175553
Last 300 lines of the log:
```console
========================================================================================
 Testing org.openjfx:javafx-base:11.0.2:13-ea+14
========================================================================================
========================================================================================
 org.openjfx:javafx-base:11.0.2:13-ea+14 stage native-image run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.openjfx:javafx-base:11.0.2
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Could not resolve all files for configuration ':testCompileClasspath'.
   > Could not find javafx-base-13-ea+14-linux.jar (org.openjfx:javafx-base:13-ea+14).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-base/13-ea+14/javafx-base-13-ea+14-linux.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2s
> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
9 actionable tasks: 1 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.openjfx:javafx-base:11.0.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 11s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 13-ea+14 failed with exit code 1.
SOFT-FAIL[native-image run][13-ea+14][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="13-ea+14" ./gradlew clean test -Pcoordinates="org.openjfx:javafx-base:11.0.2"]
========================================================================================
 org.openjfx:javafx-base:11.0.2:13-ea+14 stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 924ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
9 actionable tasks: 1 executed, 8 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Could not resolve all files for configuration ':testCompileClasspath'.
   > Could not find javafx-base-13-ea+14-linux.jar (org.openjfx:javafx-base:13-ea+14).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-base/13-ea+14/javafx-base-13-ea+14-linux.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 907ms

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'javac compile' for 13-ea+14 failed with exit code 1.
FAILED[javac compile][13-ea+14][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="13-ea+14" ./gradlew clean compileTestJava -Pcoordinates="org.openjfx:javafx-base:11.0.2"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="13-ea+14" ./gradlew clean compileTestJava -Pcoordinates="org.openjfx:javafx-base:11.0.2"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/24753645127/job/72422175597
Last 300 lines of the log:
```console
========================================================================================
 Testing org.openjfx:javafx-base:11.0.2:13-ea+14
========================================================================================
========================================================================================
 org.openjfx:javafx-base:11.0.2:13-ea+14 stage native-image run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 6s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.openjfx:javafx-base:11.0.2
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Could not resolve all files for configuration ':testCompileClasspath'.
   > Could not find javafx-base-13-ea+14-linux.jar (org.openjfx:javafx-base:13-ea+14).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-base/13-ea+14/javafx-base-13-ea+14-linux.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).

> Get more help at https://help.gradle.org.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
9 actionable tasks: 1 executed, 8 up-to-date

BUILD FAILED in 2s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.openjfx:javafx-base:11.0.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 10s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 13-ea+14 failed with exit code 1.
SOFT-FAIL[native-image run][13-ea+14][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="13-ea+14" ./gradlew clean test -Pcoordinates="org.openjfx:javafx-base:11.0.2"]
========================================================================================
 org.openjfx:javafx-base:11.0.2:13-ea+14 stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 871ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava FAILED


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
9 actionable tasks: 1 executed, 8 up-to-date
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Could not resolve all files for configuration ':testCompileClasspath'.
   > Could not find javafx-base-13-ea+14-linux.jar (org.openjfx:javafx-base:13-ea+14).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-base/13-ea+14/javafx-base-13-ea+14-linux.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 829ms

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'javac compile' for 13-ea+14 failed with exit code 1.
FAILED[javac compile][13-ea+14][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="13-ea+14" ./gradlew clean compileTestJava -Pcoordinates="org.openjfx:javafx-base:11.0.2"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="13-ea+14" ./gradlew clean compileTestJava -Pcoordinates="org.openjfx:javafx-base:11.0.2"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/24753645127/job/72422175603
Last 300 lines of the log:
```console
========================================================================================
 Testing org.openjfx:javafx-base:11.0.2:13-ea+14
========================================================================================
========================================================================================
 org.openjfx:javafx-base:11.0.2:13-ea+14 stage native-image run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 6s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.openjfx:javafx-base:11.0.2
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
9 actionable tasks: 1 executed, 8 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Could not resolve all files for configuration ':testCompileClasspath'.
   > Could not find javafx-base-13-ea+14-linux.jar (org.openjfx:javafx-base:13-ea+14).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-base/13-ea+14/javafx-base-13-ea+14-linux.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.openjfx:javafx-base:11.0.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 11s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 13-ea+14 failed with exit code 1.
SOFT-FAIL[native-image run][13-ea+14][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="13-ea+14" ./gradlew clean test -Pcoordinates="org.openjfx:javafx-base:11.0.2"]
========================================================================================
 org.openjfx:javafx-base:11.0.2:13-ea+14 stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 970ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava FAILED


FAILURE: Build failed with an exception.

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.* What went wrong:
Execution failed for task ':compileTestJava'.
> Could not resolve all files for configuration ':testCompileClasspath'.
   > Could not find javafx-base-13-ea+14-linux.jar (org.openjfx:javafx-base:13-ea+14).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-base/13-ea+14/javafx-base-13-ea+14-linux.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 932ms


For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
9 actionable tasks: 1 executed, 8 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'javac compile' for 13-ea+14 failed with exit code 1.
FAILED[javac compile][13-ea+14][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="13-ea+14" ./gradlew clean compileTestJava -Pcoordinates="org.openjfx:javafx-base:11.0.2"]
```

--- a/metadata/org.openjfx/javafx-base/13-ea/reachability-metadata.json
+++ b/metadata/org.openjfx/javafx-base/13-ea/reachability-metadata.json
@@ -1,0 +1,20 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "javafx.beans.property.adapter.JavaBeanIntegerProperty"
+      },
+      "type" : "com.sun.javafx.reflect.Trampoline",
+      "methods" : [
+        {
+          "name" : "invoke",
+          "parameterTypes" : [
+            "java.lang.reflect.Method",
+            "java.lang.Object",
+            "java.lang.Object[]"
+          ]
+        }
+      ]
+    }
+  ]
+}

--- a/metadata/org.openjfx/javafx-base/index.json
+++ b/metadata/org.openjfx/javafx-base/index.json
@@ -1,8 +1,13 @@
 [
   {
-    "latest" : true,
-    "metadata-version" : "13-ea",
-    "tested-versions" : [
+    "latest": true,
+    "metadata-version": "13-ea",
+    "source-code-url": "https://repo1.maven.org/maven2/org/openjfx/javafx-base/$version$+1/javafx-base-$version$+1-sources.jar",
+    "repository-url": "https://github.com/openjdk/jfx",
+    "test-code-url": "https://github.com/openjdk/jfx/tree/13+1/tests",
+    "documentation-url": "https://repo1.maven.org/maven2/org/openjfx/javafx-base/$version$+1/javafx-base-$version$+1-javadoc.jar",
+    "description": "JavaFX Base defines the core APIs that underpin the JavaFX toolkit, including properties, bindings, collections, and events. It provides the foundational observable and event infrastructure that higher-level JavaFX modules build on.",
+    "tested-versions": [
       "13-ea",
       "13-ea+1",
       "13-ea+10",
@@ -19,18 +24,19 @@
       "13-ea+9",
       "13"
     ],
-    "allowed-packages" : [
-      "org.openjfx"
+    "allowed-packages": [
+      "org.openjfx",
+      "javafx.beans.property.adapter"
     ]
   },
   {
-    "metadata-version" : "11.0.2",
-    "source-code-url" : "https://repo1.maven.org/maven2/org/openjfx/javafx-base/$version$/javafx-base-$version$-sources.jar",
-    "repository-url" : "https://github.com/javafxports/openjdk-jfx",
-    "test-code-url" : "https://github.com/javafxports/openjdk-jfx/tree/$version$+1/tests",
-    "documentation-url" : "https://repo1.maven.org/maven2/org/openjfx/javafx-base/$version$/javafx-base-$version$-javadoc.jar",
-    "description" : "JavaFX Base defines the core APIs that underpin the JavaFX toolkit, including properties, bindings, collections, and events. It provides the foundational observable and event infrastructure that higher-level JavaFX modules build on.",
-    "tested-versions" : [
+    "metadata-version": "11.0.2",
+    "source-code-url": "https://repo1.maven.org/maven2/org/openjfx/javafx-base/$version$/javafx-base-$version$-sources.jar",
+    "repository-url": "https://github.com/javafxports/openjdk-jfx",
+    "test-code-url": "https://github.com/javafxports/openjdk-jfx/tree/$version$+1/tests",
+    "documentation-url": "https://repo1.maven.org/maven2/org/openjfx/javafx-base/$version$/javafx-base-$version$-javadoc.jar",
+    "description": "JavaFX Base defines the core APIs that underpin the JavaFX toolkit, including properties, bindings, collections, and events. It provides the foundational observable and event infrastructure that higher-level JavaFX modules build on.",
+    "tested-versions": [
       "11.0.2",
       "12-ea+10",
       "12-ea+11",
@@ -49,7 +55,7 @@
       "12.0.1",
       "12.0.2"
     ],
-    "allowed-packages" : [
+    "allowed-packages": [
       "org.openjfx"
     ]
   }

--- a/metadata/org.openjfx/javafx-base/index.json
+++ b/metadata/org.openjfx/javafx-base/index.json
@@ -8,21 +8,7 @@
     "documentation-url": "https://repo1.maven.org/maven2/org/openjfx/javafx-base/$version$+1/javafx-base-$version$+1-javadoc.jar",
     "description": "JavaFX Base defines the core APIs that underpin the JavaFX toolkit, including properties, bindings, collections, and events. It provides the foundational observable and event infrastructure that higher-level JavaFX modules build on.",
     "tested-versions": [
-      "13-ea",
-      "13-ea+1",
-      "13-ea+10",
-      "13-ea+11",
-      "13-ea+12",
-      "13-ea+13",
-      "13-ea+2",
-      "13-ea+3",
-      "13-ea+4",
-      "13-ea+5",
-      "13-ea+6",
-      "13-ea+7",
-      "13-ea+8",
-      "13-ea+9",
-      "13"
+      "13-ea"
     ],
     "allowed-packages": [
       "org.openjfx",

--- a/metadata/org.openjfx/javafx-base/index.json
+++ b/metadata/org.openjfx/javafx-base/index.json
@@ -1,6 +1,29 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "13-ea",
+    "tested-versions" : [
+      "13-ea",
+      "13-ea+1",
+      "13-ea+10",
+      "13-ea+11",
+      "13-ea+12",
+      "13-ea+13",
+      "13-ea+2",
+      "13-ea+3",
+      "13-ea+4",
+      "13-ea+5",
+      "13-ea+6",
+      "13-ea+7",
+      "13-ea+8",
+      "13-ea+9",
+      "13"
+    ],
+    "allowed-packages" : [
+      "org.openjfx"
+    ]
+  },
+  {
     "metadata-version" : "11.0.2",
     "source-code-url" : "https://repo1.maven.org/maven2/org/openjfx/javafx-base/$version$/javafx-base-$version$-sources.jar",
     "repository-url" : "https://github.com/javafxports/openjdk-jfx",
@@ -24,21 +47,7 @@
       "12-ea+9",
       "12",
       "12.0.1",
-      "12.0.2",
-      "13-ea+1",
-      "13-ea+10",
-      "13-ea+11",
-      "13-ea+12",
-      "13-ea+13",
-      "13-ea+2",
-      "13-ea+3",
-      "13-ea+4",
-      "13-ea+5",
-      "13-ea+6",
-      "13-ea+7",
-      "13-ea+8",
-      "13-ea+9",
-      "13"
+      "12.0.2"
     ],
     "allowed-packages" : [
       "org.openjfx"

--- a/stats/org.openjfx/javafx-base/13-ea/execution-metrics.json
+++ b/stats/org.openjfx/javafx-base/13-ea/execution-metrics.json
@@ -1,0 +1,109 @@
+{
+  "fix_javac_fail:2026-05-03": {
+    "timestamp": "2026-05-03T15:05:22.277036Z",
+    "library": "org.openjfx:javafx-base:13-ea",
+    "starting_commit": "5f26ab931837bf650f6af7b757be903d879faa4f",
+    "ending_commit": "db1d9a8e7b25f76bd3c49a90c8fcc22fcd1564eb",
+    "previous_library": "org.openjfx:javafx-base:11.0.2",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "13-ea",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 0.263158,
+            "coveredCalls": 15,
+            "totalCalls": 57
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 0.275862,
+        "coveredCalls": 16,
+        "totalCalls": 58
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 5799,
+          "missed": 57692,
+          "ratio": 0.091336,
+          "total": 63491
+        },
+        "line": {
+          "covered": 1551,
+          "missed": 12809,
+          "ratio": 0.108008,
+          "total": 14360
+        },
+        "method": {
+          "covered": 460,
+          "missed": 4119,
+          "ratio": 0.100459,
+          "total": 4579
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "11.0.2",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 0.018519,
+            "coveredCalls": 1,
+            "totalCalls": 54
+          },
+          "resources": {
+            "coverageRatio": 0.0,
+            "coveredCalls": 0,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 0.018182,
+        "coveredCalls": 1,
+        "totalCalls": 55
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 5025,
+          "missed": 58430,
+          "ratio": 0.07919,
+          "total": 63455
+        },
+        "line": {
+          "covered": 1345,
+          "missed": 12940,
+          "ratio": 0.094155,
+          "total": 14285
+        },
+        "method": {
+          "covered": 411,
+          "missed": 4146,
+          "ratio": 0.090191,
+          "total": 4557
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 90350,
+      "cached_input_tokens_used": 569856,
+      "output_tokens_used": 9158,
+      "iterations": 2,
+      "cost_usd": 0.5596,
+      "tested_library_loc": 1551,
+      "metadata_entries": 2,
+      "previous_library_metadata_entries": 0,
+      "code_coverage_percent": 10.8,
+      "previous_library_coverage_percent": 9.42
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.openjfx/javafx-base/13-ea/src/test/java/org_openjfx/javafx_base/PropertyDescriptorTest.java",
+      "metadata_file": "metadata/org.openjfx/javafx-base/13-ea/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.openjfx/javafx-base/13-ea/stats.json
+++ b/stats/org.openjfx/javafx-base/13-ea/stats.json
@@ -1,0 +1,42 @@
+{
+  "versions" : [ {
+    "version" : "13-ea",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 0.263158,
+          "coveredCalls" : 15,
+          "totalCalls" : 57
+        },
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 1,
+          "totalCalls" : 1
+        }
+      },
+      "coverageRatio" : 0.275862,
+      "coveredCalls" : 16,
+      "totalCalls" : 58
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 5799,
+        "missed" : 57692,
+        "ratio" : 0.091336,
+        "total" : 63491
+      },
+      "line" : {
+        "covered" : 1551,
+        "missed" : 12809,
+        "ratio" : 0.108008,
+        "total" : 14360
+      },
+      "method" : {
+        "covered" : 460,
+        "missed" : 4119,
+        "ratio" : 0.100459,
+        "total" : 4579
+      }
+    }
+  } ]
+}

--- a/tests/src/org.openjfx/javafx-base/13-ea/.gitignore
+++ b/tests/src/org.openjfx/javafx-base/13-ea/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.openjfx/javafx-base/13-ea/build.gradle
+++ b/tests/src/org.openjfx/javafx-base/13-ea/build.gradle
@@ -1,0 +1,29 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+String osName = System.getProperty("os.name").toLowerCase()
+String javafxPlatform = osName.contains("win") ? "win" : osName.contains("mac") ? "mac" : "linux"
+
+dependencies {
+    testImplementation "org.openjfx:javafx-base:$libraryVersion:$javafxPlatform"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.openjfx/javafx-base/13-ea/build.gradle
+++ b/tests/src/org.openjfx/javafx-base/13-ea/build.gradle
@@ -9,11 +9,12 @@ plugins {
 }
 
 String libraryVersion = tck.testedLibraryVersion.get()
+String dependencyVersion = libraryVersion == "13-ea" ? "13-ea+1" : libraryVersion
 String osName = System.getProperty("os.name").toLowerCase()
 String javafxPlatform = osName.contains("win") ? "win" : osName.contains("mac") ? "mac" : "linux"
 
 dependencies {
-    testImplementation "org.openjfx:javafx-base:$libraryVersion:$javafxPlatform"
+    testImplementation "org.openjfx:javafx-base:$dependencyVersion:$javafxPlatform"
     testImplementation 'org.assertj:assertj-core:3.22.0'
 }
 

--- a/tests/src/org.openjfx/javafx-base/13-ea/gradle.properties
+++ b/tests/src/org.openjfx/javafx-base/13-ea/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.openjfx:javafx-base:13-ea
+library.version = 13-ea
+metadata.dir = org.openjfx/javafx-base/13-ea/

--- a/tests/src/org.openjfx/javafx-base/13-ea/settings.gradle
+++ b/tests/src/org.openjfx/javafx-base/13-ea/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.openjfx.javafx-base_tests'

--- a/tests/src/org.openjfx/javafx-base/13-ea/src/test/java/org_openjfx/javafx_base/Javafx_baseTest.java
+++ b/tests/src/org.openjfx/javafx-base/13-ea/src/test/java/org_openjfx/javafx_base/Javafx_baseTest.java
@@ -1,0 +1,344 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_openjfx.javafx_base;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import javafx.beans.Observable;
+import javafx.beans.binding.Bindings;
+import javafx.beans.binding.StringBinding;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.property.SimpleIntegerProperty;
+import javafx.beans.property.SimpleListProperty;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+import javafx.collections.ArrayChangeListener;
+import javafx.collections.FXCollections;
+import javafx.collections.ListChangeListener;
+import javafx.collections.MapChangeListener;
+import javafx.collections.ObservableIntegerArray;
+import javafx.collections.ObservableList;
+import javafx.collections.ObservableMap;
+import javafx.collections.ObservableSet;
+import javafx.collections.SetChangeListener;
+import javafx.collections.transformation.FilteredList;
+import javafx.collections.transformation.SortedList;
+import javafx.event.Event;
+import javafx.event.EventDispatchChain;
+import javafx.event.EventDispatcher;
+import javafx.event.EventTarget;
+import javafx.event.EventType;
+import javafx.util.Duration;
+import org.junit.jupiter.api.Test;
+
+public class Javafx_baseTest {
+
+    private static final EventType<Event> PROFILE_UPDATED_EVENT = new EventType<>(Event.ANY, "PROFILE_UPDATED_EVENT");
+
+    @Test
+    void propertiesAndBindingsStayInSyncAcrossBindingModes() {
+        SimpleStringProperty primaryName = new SimpleStringProperty("Ada");
+        SimpleStringProperty nickname = new SimpleStringProperty("Countess");
+        SimpleStringProperty mirror = new SimpleStringProperty("Ada");
+        SimpleBooleanProperty includeNickname = new SimpleBooleanProperty(true);
+        StringBinding displayName = Bindings.createStringBinding(
+                () -> includeNickname.get() ? primaryName.get() + " (" + nickname.get() + ")" : primaryName.get(),
+                primaryName,
+                nickname,
+                includeNickname);
+
+        try {
+            List<String> displayChanges = new ArrayList<>();
+            displayName.addListener((observable, oldValue, newValue) -> displayChanges.add(oldValue + "->" + newValue));
+            Bindings.bindBidirectional(primaryName, mirror);
+
+            assertThat(displayName.get()).isEqualTo("Ada (Countess)");
+
+            mirror.set("Grace");
+            assertThat(primaryName.get()).isEqualTo("Grace");
+            assertThat(displayName.get()).isEqualTo("Grace (Countess)");
+
+            includeNickname.set(false);
+            assertThat(displayName.get()).isEqualTo("Grace");
+
+            primaryName.set("Barbara");
+            assertThat(mirror.get()).isEqualTo("Barbara");
+            assertThat(displayName.get()).isEqualTo("Barbara");
+
+            nickname.set("Compiler");
+            includeNickname.set(true);
+            assertThat(displayName.get()).isEqualTo("Barbara (Compiler)");
+
+            Bindings.unbindBidirectional(primaryName, mirror);
+            mirror.set("Detached");
+            assertThat(primaryName.get()).isEqualTo("Barbara");
+            assertThat(displayChanges).isNotEmpty();
+            assertThat(displayChanges.get(displayChanges.size() - 1)).isEqualTo("Barbara->Barbara (Compiler)");
+        } finally {
+            displayName.dispose();
+        }
+    }
+
+    @Test
+    void observableIntegerArrayTracksChangesAndCopiesValues() {
+        ObservableIntegerArray values = FXCollections.observableIntegerArray(1, 2, 3);
+        List<String> arrayChanges = new ArrayList<>();
+
+        values.addListener((ArrayChangeListener<ObservableIntegerArray>) (observableArray, sizeChanged, from, to) ->
+                arrayChanges.add(sizeChanged + ":" + from + "-" + to));
+
+        values.set(1, 5);
+        values.addAll(8, 13);
+        values.resize(4);
+
+        ObservableIntegerArray copy = FXCollections.observableIntegerArray(values);
+        values.set(0, 21);
+
+        assertThat(values.toArray(null)).containsExactly(21, 5, 3, 8);
+        assertThat(copy.toArray(null)).containsExactly(1, 5, 3, 8);
+        assertThat(arrayChanges).hasSize(4);
+        assertThat(arrayChanges.get(0)).startsWith("false:1-");
+        assertThat(arrayChanges.get(1)).startsWith("true:3-");
+    }
+
+    @Test
+    void filteredAndSortedListsReactToSourceAndElementChanges() {
+        ObservableList<TaskItem> tasks = FXCollections.observableArrayList(task -> new Observable[] {
+                task.nameProperty(),
+                task.priorityProperty()
+        });
+        TaskItem beta = new TaskItem("beta", 2);
+        TaskItem alpha = new TaskItem("alpha", 1);
+        TaskItem gamma = new TaskItem("gamma", 3);
+        tasks.addAll(beta, alpha, gamma);
+
+        FilteredList<TaskItem> prioritized = new FilteredList<>(tasks, task -> task.getPriority() >= 2);
+        SortedList<TaskItem> sorted = new SortedList<>(prioritized, Comparator.comparing(TaskItem::getName));
+        List<List<String>> snapshots = new ArrayList<>();
+        sorted.addListener((ListChangeListener<TaskItem>) change -> {
+            while (change.next()) {
+                // exhaust change details so the transformation is fully applied
+            }
+            snapshots.add(taskNames(sorted));
+        });
+
+        assertThat(taskNames(prioritized)).containsExactly("beta", "gamma");
+        assertThat(taskNames(sorted)).containsExactly("beta", "gamma");
+
+        alpha.setPriority(4);
+        assertThat(taskNames(prioritized)).containsExactly("beta", "alpha", "gamma");
+        assertThat(taskNames(sorted)).containsExactly("alpha", "beta", "gamma");
+
+        gamma.setName("aardvark");
+        assertThat(taskNames(sorted)).containsExactly("aardvark", "alpha", "beta");
+        assertThat(prioritized.getSourceIndex(1)).isEqualTo(1);
+        assertThat(snapshots).contains(List.of("alpha", "beta", "gamma"), List.of("aardvark", "alpha", "beta"));
+    }
+
+    @Test
+    void observableMapAndSetListenersReportMutations() {
+        ObservableMap<String, Integer> counts = FXCollections.observableHashMap();
+        ObservableSet<String> labels = FXCollections.observableSet("base");
+        List<String> mapEvents = new ArrayList<>();
+        List<String> setEvents = new ArrayList<>();
+
+        counts.addListener((MapChangeListener<String, Integer>) change -> {
+            if (change.wasAdded() && change.wasRemoved()) {
+                mapEvents.add("replaced " + change.getKey() + ":" + change.getValueRemoved() + "->" + change.getValueAdded());
+            } else if (change.wasAdded()) {
+                mapEvents.add("added " + change.getKey() + ":" + change.getValueAdded());
+            } else {
+                mapEvents.add("removed " + change.getKey() + ":" + change.getValueRemoved());
+            }
+        });
+        labels.addListener((SetChangeListener<String>) change -> {
+            if (change.wasAdded()) {
+                setEvents.add("added " + change.getElementAdded());
+            }
+            if (change.wasRemoved()) {
+                setEvents.add("removed " + change.getElementRemoved());
+            }
+        });
+
+        counts.put("listeners", 1);
+        counts.put("listeners", 2);
+        counts.remove("listeners");
+        labels.add("bindings");
+        labels.remove("base");
+
+        assertThat(mapEvents).containsExactly(
+                "added listeners:1",
+                "replaced listeners:1->2",
+                "removed listeners:2");
+        assertThat(setEvents).containsExactly("added bindings", "removed base");
+        assertThat(labels).containsExactly("bindings");
+    }
+
+    @Test
+    void eventDispatchChainUsesCustomDispatchersAndSupportsCopyFor() {
+        List<String> order = new ArrayList<>();
+        RecordingEventTarget target = new RecordingEventTarget(
+                new RecordingDispatcher("first", order, false),
+                new RecordingDispatcher("second", order, true),
+                new RecordingDispatcher("third", order, false));
+
+        Event original = new Event("source", target, PROFILE_UPDATED_EVENT);
+        original.consume();
+
+        Event copied = original.copyFor("copy", target);
+        assertThat(copied.getSource()).isEqualTo("copy");
+        assertThat(copied.getTarget()).isSameAs(target);
+        assertThat(copied.getEventType()).isSameAs(PROFILE_UPDATED_EVENT);
+        assertThat(copied.isConsumed()).isFalse();
+
+        Event.fireEvent(target, copied);
+
+        assertThat(order).containsExactly("first:PROFILE_UPDATED_EVENT", "second:PROFILE_UPDATED_EVENT");
+        assertThat(copied.isConsumed()).isTrue();
+    }
+
+    @Test
+    void listPropertyTracksContentSizeAndAssignedLists() {
+        SimpleListProperty<String> names = new SimpleListProperty<>(FXCollections.observableArrayList("Ada"));
+        List<String> sizeChanges = new ArrayList<>();
+        List<List<String>> snapshots = new ArrayList<>();
+        List<Boolean> emptyStates = new ArrayList<>();
+
+        names.sizeProperty().addListener((observable, oldValue, newValue) -> sizeChanges.add(oldValue + "->" + newValue));
+        names.addListener((ListChangeListener<String>) change -> {
+            while (change.next()) {
+                // exhaust change details so the property state is fully applied
+            }
+            snapshots.add(List.copyOf(names));
+        });
+        names.emptyProperty().addListener((observable, oldValue, newValue) -> emptyStates.add(newValue));
+
+        names.addAll("Grace", "Barbara");
+        names.set(FXCollections.observableArrayList("Diana", "Evelyn"));
+        names.remove("Diana");
+        names.clear();
+
+        assertThat(names).isEmpty();
+        assertThat(sizeChanges).containsExactly("1->3", "3->2", "2->1", "1->0");
+        assertThat(snapshots).containsExactly(
+                List.of("Ada", "Grace", "Barbara"),
+                List.of("Diana", "Evelyn"),
+                List.of("Evelyn"),
+                List.of());
+        assertThat(emptyStates).containsExactly(true);
+    }
+
+    @Test
+    void durationFactoriesParsingAndArithmeticRemainConsistent() {
+        Duration parsedMilliseconds = Duration.valueOf("1500ms");
+        Duration parsedSeconds = Duration.valueOf("2.5s");
+        Duration scaledMinute = Duration.minutes(1).divide(4);
+        Duration combined = parsedMilliseconds.add(parsedSeconds).subtract(Duration.seconds(1));
+
+        assertThat(parsedMilliseconds).isEqualTo(Duration.millis(1500));
+        assertThat(parsedSeconds).isEqualTo(Duration.millis(2500));
+        assertThat(scaledMinute).isEqualTo(Duration.seconds(15));
+        assertThat(combined).isEqualTo(Duration.seconds(3));
+        assertThat(combined.greaterThan(Duration.seconds(2))).isTrue();
+        assertThat(combined.lessThan(Duration.seconds(4))).isTrue();
+    }
+
+    @Test
+    void durationSentinelValuesPreserveSpecialStateAcrossOperations() {
+        Duration indefinite = Duration.INDEFINITE.add(Duration.seconds(5)).divide(3);
+        Duration unknown = Duration.UNKNOWN.multiply(2).negate();
+
+        assertThat(indefinite.isIndefinite()).isTrue();
+        assertThat(indefinite.greaterThan(Duration.hours(1))).isTrue();
+        assertThat(unknown.isUnknown()).isTrue();
+        assertThat(unknown.compareTo(Duration.ZERO)).isEqualTo(1);
+    }
+
+    private static List<String> taskNames(List<TaskItem> tasks) {
+        List<String> names = new ArrayList<>(tasks.size());
+        for (TaskItem task : tasks) {
+            names.add(task.getName());
+        }
+        return names;
+    }
+
+    public static final class TaskItem {
+        private final StringProperty name = new SimpleStringProperty(this, "name");
+        private final SimpleIntegerProperty priority = new SimpleIntegerProperty(this, "priority");
+
+        public TaskItem(String name, int priority) {
+            this.name.set(name);
+            this.priority.set(priority);
+        }
+
+        public String getName() {
+            return name.get();
+        }
+
+        public void setName(String name) {
+            this.name.set(name);
+        }
+
+        public StringProperty nameProperty() {
+            return name;
+        }
+
+        public int getPriority() {
+            return priority.get();
+        }
+
+        public void setPriority(int priority) {
+            this.priority.set(priority);
+        }
+
+        public SimpleIntegerProperty priorityProperty() {
+            return priority;
+        }
+    }
+
+    public static final class RecordingEventTarget implements EventTarget {
+        private final List<EventDispatcher> dispatchers;
+
+        public RecordingEventTarget(EventDispatcher... dispatchers) {
+            this.dispatchers = List.of(dispatchers);
+        }
+
+        @Override
+        public EventDispatchChain buildEventDispatchChain(EventDispatchChain tail) {
+            EventDispatchChain chain = tail;
+            for (EventDispatcher dispatcher : dispatchers) {
+                chain = chain.append(dispatcher);
+            }
+            return chain;
+        }
+    }
+
+    public static final class RecordingDispatcher implements EventDispatcher {
+        private final String name;
+        private final List<String> order;
+        private final boolean consume;
+
+        public RecordingDispatcher(String name, List<String> order, boolean consume) {
+            this.name = name;
+            this.order = order;
+            this.consume = consume;
+        }
+
+        @Override
+        public Event dispatchEvent(Event event, EventDispatchChain tail) {
+            order.add(name + ":" + event.getEventType().getName());
+            if (consume) {
+                event.consume();
+                return event;
+            }
+            return tail.dispatchEvent(event);
+        }
+    }
+}

--- a/tests/src/org.openjfx/javafx-base/13-ea/src/test/java/org_openjfx/javafx_base/PropertyDescriptorTest.java
+++ b/tests/src/org.openjfx/javafx-base/13-ea/src/test/java/org_openjfx/javafx_base/PropertyDescriptorTest.java
@@ -1,0 +1,112 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_openjfx.javafx_base;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.beans.VetoableChangeListener;
+import java.util.ArrayList;
+import java.util.List;
+import javafx.beans.property.adapter.JavaBeanIntegerProperty;
+import javafx.beans.property.adapter.JavaBeanIntegerPropertyBuilder;
+import org.junit.jupiter.api.Test;
+
+public class PropertyDescriptorTest {
+
+    @Test
+    void javaBeanPropertyRegistersAndRemovesNamedVetoableChangeListener() throws NoSuchMethodException {
+        NamedVetoableBean bean = new NamedVetoableBean(7);
+
+        JavaBeanIntegerProperty property = JavaBeanIntegerPropertyBuilder.create()
+                .bean(bean)
+                .name("score")
+                .build();
+
+        assertThat(property.get()).isEqualTo(7);
+        assertThat(bean.addedPropertyNames).containsExactly("score");
+        assertThat(bean.vetoableChangeListeners).hasSize(1);
+
+        property.dispose();
+
+        assertThat(bean.removedPropertyNames).containsExactly("score");
+        assertThat(bean.vetoableChangeListeners).isEmpty();
+    }
+
+    @Test
+    void javaBeanPropertyRegistersAndRemovesGlobalVetoableChangeListener() throws NoSuchMethodException {
+        GlobalVetoableBean bean = new GlobalVetoableBean(13);
+
+        JavaBeanIntegerProperty property = JavaBeanIntegerPropertyBuilder.create()
+                .bean(bean)
+                .name("score")
+                .build();
+
+        assertThat(property.get()).isEqualTo(13);
+        assertThat(bean.addedListeners).hasSize(1);
+
+        property.dispose();
+
+        assertThat(bean.removedListeners).hasSize(1);
+        assertThat(bean.addedListeners).isEmpty();
+    }
+
+    public static final class NamedVetoableBean {
+        private final List<String> addedPropertyNames = new ArrayList<>();
+        private final List<String> removedPropertyNames = new ArrayList<>();
+        private final List<VetoableChangeListener> vetoableChangeListeners = new ArrayList<>();
+        private int score;
+
+        public NamedVetoableBean(int score) {
+            this.score = score;
+        }
+
+        public int getScore() {
+            return score;
+        }
+
+        public void setScore(int score) {
+            this.score = score;
+        }
+
+        public void addVetoableChangeListener(String propertyName, VetoableChangeListener listener) {
+            addedPropertyNames.add(propertyName);
+            vetoableChangeListeners.add(listener);
+        }
+
+        public void removeVetoableChangeListener(String propertyName, VetoableChangeListener listener) {
+            removedPropertyNames.add(propertyName);
+            vetoableChangeListeners.remove(listener);
+        }
+    }
+
+    public static final class GlobalVetoableBean {
+        private final List<VetoableChangeListener> addedListeners = new ArrayList<>();
+        private final List<VetoableChangeListener> removedListeners = new ArrayList<>();
+        private int score;
+
+        public GlobalVetoableBean(int score) {
+            this.score = score;
+        }
+
+        public int getScore() {
+            return score;
+        }
+
+        public void setScore(int score) {
+            this.score = score;
+        }
+
+        public void addVetoableChangeListener(VetoableChangeListener listener) {
+            addedListeners.add(listener);
+        }
+
+        public void removeVetoableChangeListener(VetoableChangeListener listener) {
+            removedListeners.add(listener);
+            addedListeners.remove(listener);
+        }
+    }
+}

--- a/tests/src/org.openjfx/javafx-base/13-ea/user-code-filter.json
+++ b/tests/src/org.openjfx/javafx-base/13-ea/user-code-filter.json
@@ -1,0 +1,10 @@
+{
+  "rules": [
+    {
+      "excludeClasses": "**"
+    },
+    {
+      "includeClasses": "org.openjfx.**"
+    }
+  ]
+}

--- a/tests/tck-build-logic/src/main/groovy/org/graalvm/internal/tck/harness/tasks/PopulateArtifactURLs.java
+++ b/tests/tck-build-logic/src/main/groovy/org/graalvm/internal/tck/harness/tasks/PopulateArtifactURLs.java
@@ -271,8 +271,9 @@ public class PopulateArtifactURLs extends DefaultTask {
         String indexFile = target.indexPath.toString().replace('\\', '/');
         String metadataVersion = target.entry.metadataVersion();
         String testVersion = hasText(target.entry.testVersion()) ? target.entry.testVersion() : metadataVersion;
-        String urlUpdateInstructions = urlUpdateInstructions(overwriteExisting, metadataVersion);
-        String sourceArtifactVerificationInstructions = sourceArtifactVerificationInstructions(verifyArtifactSources, metadataVersion);
+        String testedVersions = testedVersionsDisplay(target.entry.testedVersions(), metadataVersion);
+        String urlUpdateInstructions = urlUpdateInstructions(overwriteExisting);
+        String sourceArtifactVerificationInstructions = sourceArtifactVerificationInstructions(verifyArtifactSources, testedVersions);
 
         return """
                 Find the repository URL, the sources URL, the test suite URL, the documentation URL, and a concise two-sentence explanation for the following library: %s:%s:%s
@@ -282,7 +283,7 @@ public class PopulateArtifactURLs extends DefaultTask {
                 - { "name": "scala", "version": "2" } for Scala 2 libraries
                 - { "name": "scala", "version": "3" } for Scala 3 libraries
                 If the library is not language-specific, leave the "language" field absent.
-                The sources URL, the test suite URL, and the documentation URL must render to the entry metadata-version "%s" when "$version$" is replaced with "%s".
+                The sources URL, the test suite URL, and the documentation URL must render correctly for every entry tested-version alias when "$version$" is replaced with that alias. Entry tested-versions: [%s].
                 The source, test suite, and documentation URLs should point to the right tag of the library.
                 If we have these source of these artifacts on maven, that should be prefered.
                 If there are tests on maven that should also be prefered.
@@ -312,12 +313,12 @@ public class PopulateArtifactURLs extends DefaultTask {
                 - Coordinate version: %s
                 - Entry metadata-version: %s
                 - Entry test-version: %s
+                - Entry tested-versions: [%s]
                 """.formatted(
                 target.group,
                 target.artifact,
                 target.version,
-                metadataVersion,
-                metadataVersion,
+                testedVersions,
                 sourceArtifactVerificationInstructions,
                 indexFile,
                 metadataVersion,
@@ -330,58 +331,66 @@ public class PopulateArtifactURLs extends DefaultTask {
                 currentLanguageValue(target.entry),
                 target.version,
                 metadataVersion,
-                testVersion
+                testVersion,
+                testedVersions
         ).strip();
     }
 
-    private static String sourceArtifactVerificationInstructions(boolean verifyArtifactSources, String version) {
+    private static String sourceArtifactVerificationInstructions(boolean verifyArtifactSources, String testedVersions) {
         if (!verifyArtifactSources) {
             return "";
         }
         return """
                 Source Artifact Verification (required):
-                - Verify candidate source URLs for version "%s", including Maven and non-Maven candidates.
+                - Verify candidate source URLs for every entry tested-version alias ([%s]), including Maven and non-Maven candidates.
                 - If you use Maven source artifacts, confirm `-sources.jar` contains real source files (`.java`, `.kt`, `.scala`, `.groovy`), not only metadata/license files.
                 - If you use Maven test-source artifacts, confirm `-test-sources.jar` contains real test source files.
                 - For non-Maven source/test URLs (for example repository tree pages or downloadable archives), verify that the selected URL resolves to real source/test files for the exact version.
                 - If a candidate source/test URL fails this verification, do not use it. Prefer a verified repository tag URL instead.
-                - If an existing URL can be rendered as a version template, verify the rendered exact-version candidate before writing the "$version$" template.
-                """.formatted(version).strip();
+                - If an existing URL can be rendered as a version template, verify the rendered candidate for every entry tested-version alias before writing the "$version$" template.
+                """.formatted(testedVersions).strip();
     }
 
-    private static String urlUpdateInstructions(boolean overwriteExisting, String version) {
+    private static String urlUpdateInstructions(boolean overwriteExisting) {
         if (overwriteExisting) {
             return """
                     - Overwrite existing URL values.
-                    - Set "source-code-url" to the selected source URL with "$version$" replacing version "%s".
+                    - Set "source-code-url" to the selected source URL with "$version$" replacing each tested-version alias only when that template is valid for every alias.
                     - Set "repository-url" to the selected repository URL.
                     - "repository-url" must be the canonical repository root URL and must not include a version/tag/branch path (for example, no "/tree/v_1.2.11").
-                    - Set "test-code-url" to the selected test suite URL with "$version$" replacing version "%s".
-                    - Set "documentation-url" to the selected project documentation URL with "$version$" replacing version "%s".
+                    - Set "test-code-url" to the selected test suite URL with "$version$" replacing each tested-version alias only when that template is valid for every alias.
+                    - Set "documentation-url" to the selected project documentation URL with "$version$" replacing each tested-version alias only when that template is valid for every alias.
                     - Treat existing versioned "source-code-url", "test-code-url", and "documentation-url" values as templates when they contain the current entry version or another artifact version.
-                    - Render template candidates for entry metadata-version "%s" and verify rendered URLs before writing them.
+                    - Render template candidates for every entry tested-version alias and verify rendered URLs before writing them.
                     - If template verification fails, search for a correct exact-version URL instead of preserving a stale value.
                     - A non-empty URL pointing at a different artifact version is not considered already correct when URL maintenance is requested.
                     - Set "description" to a concise explanation of the library in exactly two sentences.
                     - Set "language" to the structured language object when the library is language-specific; otherwise leave the field absent.
-                    """.formatted(version, version, version, version).strip();
+                    """.strip();
         }
         return """
                 - Fill only missing fields among "source-code-url", "repository-url", "test-code-url", "documentation-url", "description", and "language".
                 - A field is missing only when absent, null, or blank.
                 - Do not modify fields that already contain a non-blank value.
-                - Set missing "source-code-url" to the selected source URL with "$version$" replacing version "%s".
+                - Set missing "source-code-url" to the selected source URL with "$version$" replacing each tested-version alias only when that template is valid for every alias.
                 - Set missing "repository-url" to the selected repository URL.
                 - "repository-url" must be the canonical repository root URL and must not include a version/tag/branch path (for example, no "/tree/v_1.2.11").
-                - Set missing "test-code-url" to the selected test suite URL with "$version$" replacing version "%s".
-                - Set missing "documentation-url" to the selected project documentation URL with "$version$" replacing version "%s".
+                - Set missing "test-code-url" to the selected test suite URL with "$version$" replacing each tested-version alias only when that template is valid for every alias.
+                - Set missing "documentation-url" to the selected project documentation URL with "$version$" replacing each tested-version alias only when that template is valid for every alias.
                 - When URL maintenance is requested, treat existing versioned "source-code-url", "test-code-url", and "documentation-url" values as templates when they contain the current entry version or another artifact version.
-                - Render template candidates for entry metadata-version "%s" and verify rendered URLs before writing them.
+                - Render template candidates for every entry tested-version alias and verify rendered URLs before writing them.
                 - If template verification fails, search for a correct exact-version URL instead of preserving a stale value.
                 - A non-empty URL pointing at a different artifact version is not considered already correct when URL maintenance is requested.
                 - Set missing "description" to a concise explanation of the library in exactly two sentences.
                 - Set missing "language" only when the library is language-specific; otherwise leave the field absent.
-                """.formatted(version, version, version, version).strip();
+                """.strip();
+    }
+
+    private static String testedVersionsDisplay(List<String> testedVersions, String metadataVersion) {
+        if (testedVersions == null || testedVersions.isEmpty()) {
+            return metadataVersion;
+        }
+        return String.join(", ", testedVersions);
     }
 
     private static String currentValue(String value) {

--- a/tests/tck-build-logic/src/main/groovy/org/graalvm/internal/tck/harness/tasks/ValidateIndexFilesTask.java
+++ b/tests/tck-build-logic/src/main/groovy/org/graalvm/internal/tck/harness/tasks/ValidateIndexFilesTask.java
@@ -22,13 +22,16 @@ import org.jetbrains.annotations.NotNull;
 import org.gradle.util.internal.VersionNumber;
 
 import java.io.File;
+import java.net.URI;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.LinkedHashSet;
 import java.util.List;
+import java.util.Locale;
 import java.util.Map;
 import java.util.Set;
+import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 /**
@@ -46,6 +49,17 @@ import java.util.regex.Pattern;
 public abstract class ValidateIndexFilesTask extends CoordinatesAwareTask {
 
     private static final Pattern METADATA_PATTERN = Pattern.compile("metadata/[^/]+/[^/]+/index\\.json");
+    private static final Pattern METADATA_COORDINATE_PATTERN = Pattern.compile("metadata/([^/]+)/([^/]+)/index\\.json");
+    private static final String VERSION_TOKEN = "$version$";
+    private static final List<String> URL_TEMPLATE_FIELDS = List.of(
+            "source-code-url",
+            "test-code-url",
+            "documentation-url"
+    );
+    private static final Set<String> MAVEN_CENTRAL_HOSTS = Set.of(
+            "repo1.maven.org",
+            "repo.maven.apache.org"
+    );
 
     @Input
     @Optional
@@ -152,6 +166,7 @@ public abstract class ValidateIndexFilesTask extends CoordinatesAwareTask {
                 // Additional semantic validations for library metadata index files
                 if (METADATA_PATTERN.matcher(filePath).matches()) {
                     checkLibraryIndexTestedVersions(json, filePath, failures);
+                    checkLibraryIndexUrlTemplates(json, filePath, failures);
                 }
 
                 // Print success only if no new failures were added by schema or semantic checks
@@ -167,6 +182,114 @@ public abstract class ValidateIndexFilesTask extends CoordinatesAwareTask {
             failures.forEach(f -> getLogger().error(f));
             throw new GradleException("Validation failed for " + failures.size() + " file(s).");
         }
+    }
+
+    /**
+     * Checks Maven Central URL templates against every tested-version alias in the entry.
+     * <p>
+     * This is intentionally a structural check instead of an HTTP probe so the global
+     * index validator remains deterministic. It catches templates such as
+     * {@code $version$+1} on entries that also contain aliases like {@code 13} or
+     * {@code 13-ea+1}, where rendering creates untracked Maven artifact versions.
+     */
+    static void checkLibraryIndexUrlTemplates(JsonNode json, String filePath, List<String> failures) {
+        if (json == null || !json.isArray()) {
+            return;
+        }
+
+        Matcher coordinateMatcher = METADATA_COORDINATE_PATTERN.matcher(filePath);
+        if (!coordinateMatcher.matches()) {
+            return;
+        }
+
+        String groupPath = coordinateMatcher.group(1).replace('.', '/');
+        String artifact = coordinateMatcher.group(2);
+
+        for (JsonNode entry : json) {
+            String metadataVersion = entry.path("metadata-version").isTextual()
+                    ? entry.path("metadata-version").asText()
+                    : "<unknown>";
+            List<String> testedVersions = testedVersions(entry);
+            if (testedVersions.size() <= 1) {
+                continue;
+            }
+
+            Set<String> testedVersionSet = new LinkedHashSet<>(testedVersions);
+            for (String fieldName : URL_TEMPLATE_FIELDS) {
+                JsonNode urlNode = entry.get(fieldName);
+                if (urlNode == null || !urlNode.isTextual()) {
+                    continue;
+                }
+
+                String template = urlNode.asText();
+                if (!template.contains(VERSION_TOKEN)) {
+                    continue;
+                }
+
+                for (String testedVersion : testedVersions) {
+                    String renderedUrl = template.replace(VERSION_TOKEN, testedVersion);
+                    String renderedMavenVersion = mavenCentralArtifactVersion(renderedUrl, groupPath, artifact);
+                    if (renderedMavenVersion == null || testedVersion.equals(renderedMavenVersion)) {
+                        continue;
+                    }
+                    if (!testedVersionSet.contains(renderedMavenVersion)) {
+                        failures.add("❌ " + filePath + ": " + fieldName
+                                + " template under metadata-version " + metadataVersion
+                                + " renders tested-version " + testedVersion
+                                + " to Maven artifact version " + renderedMavenVersion
+                                + ", which is not listed in tested-versions. Split the entry or use an alias-safe URL.");
+                    }
+                }
+            }
+        }
+    }
+
+    private static List<String> testedVersions(JsonNode entry) {
+        JsonNode testedVersionsNode = entry.get("tested-versions");
+        if (testedVersionsNode == null || !testedVersionsNode.isArray()) {
+            return Collections.emptyList();
+        }
+
+        List<String> testedVersions = new ArrayList<>();
+        for (JsonNode testedVersionNode : testedVersionsNode) {
+            if (testedVersionNode != null && testedVersionNode.isTextual()) {
+                testedVersions.add(testedVersionNode.asText());
+            }
+        }
+        return testedVersions;
+    }
+
+    private static String mavenCentralArtifactVersion(String renderedUrl, String groupPath, String artifact) {
+        URI uri;
+        try {
+            uri = URI.create(renderedUrl);
+        } catch (IllegalArgumentException e) {
+            return null;
+        }
+
+        String host = uri.getHost();
+        if (host == null || !MAVEN_CENTRAL_HOSTS.contains(host.toLowerCase(Locale.ROOT))) {
+            return null;
+        }
+
+        String expectedPrefix = "/maven2/" + groupPath + "/" + artifact + "/";
+        String path = uri.getPath();
+        if (path == null || !path.startsWith(expectedPrefix)) {
+            return null;
+        }
+
+        String remainder = path.substring(expectedPrefix.length());
+        int separator = remainder.indexOf('/');
+        if (separator <= 0 || separator == remainder.length() - 1) {
+            return null;
+        }
+
+        String artifactVersion = remainder.substring(0, separator);
+        String fileName = remainder.substring(separator + 1);
+        if (!fileName.startsWith(artifact + "-" + artifactVersion + "-")) {
+            return null;
+        }
+        return artifactVersion;
     }
 
     /**

--- a/tests/tck-build-logic/src/test/java/org/graalvm/internal/tck/harness/tasks/PopulateArtifactURLsTests.java
+++ b/tests/tck-build-logic/src/test/java/org/graalvm/internal/tck/harness/tasks/PopulateArtifactURLsTests.java
@@ -53,14 +53,14 @@ class PopulateArtifactURLsTests {
                 .contains("{ \"name\": \"groovy\", \"version\": \"<groovy major.minor, e.g. 4.0>\" }")
                 .contains("{ \"name\": \"scala\", \"version\": \"2\" }")
                 .contains("If the library is not language-specific, leave the \"language\" field absent.")
-                .contains("The sources URL, the test suite URL, and the documentation URL must render to the entry metadata-version \"1.4.1\" when \"$version$\" is replaced with \"1.4.1\".")
+                .contains("The sources URL, the test suite URL, and the documentation URL must render correctly for every entry tested-version alias when \"$version$\" is replaced with that alias. Entry tested-versions: [1.4.1].")
                 .contains("The \"description\" field must explain what the library does in exactly two sentences.")
                 .contains("Fill only missing fields among \"source-code-url\", \"repository-url\", \"test-code-url\", \"documentation-url\", \"description\", and \"language\".")
                 .contains("Set missing \"repository-url\" to the selected repository URL.")
                 .contains("\"repository-url\" must be the canonical repository root URL and must not include a version/tag/branch path (for example, no \"/tree/v_1.2.11\").")
-                .contains("Set missing \"source-code-url\" to the selected source URL with \"$version$\" replacing version \"1.4.1\".")
-                .contains("Set missing \"documentation-url\" to the selected project documentation URL with \"$version$\" replacing version \"1.4.1\".")
-                .contains("Render template candidates for entry metadata-version \"1.4.1\" and verify rendered URLs before writing them.")
+                .contains("Set missing \"source-code-url\" to the selected source URL with \"$version$\" replacing each tested-version alias only when that template is valid for every alias.")
+                .contains("Set missing \"documentation-url\" to the selected project documentation URL with \"$version$\" replacing each tested-version alias only when that template is valid for every alias.")
+                .contains("Render template candidates for every entry tested-version alias and verify rendered URLs before writing them.")
                 .contains("A non-empty URL pointing at a different artifact version is not considered already correct when URL maintenance is requested.")
                 .contains("Set missing \"description\" to a concise explanation of the library in exactly two sentences.")
                 .contains("Set missing \"language\" only when the library is language-specific; otherwise leave the field absent.")
@@ -178,16 +178,16 @@ class PopulateArtifactURLsTests {
         String output = outputBuffer.toString(StandardCharsets.UTF_8);
         assertThat(output)
                 .contains("Source Artifact Verification (required):")
-                .contains("Verify candidate source URLs for version \"1.4.1\", including Maven and non-Maven candidates.")
+                .contains("Verify candidate source URLs for every entry tested-version alias ([1.4.1]), including Maven and non-Maven candidates.")
                 .contains("confirm `-sources.jar` contains real source files")
                 .contains("confirm `-test-sources.jar` contains real test source files.")
                 .contains("For non-Maven source/test URLs")
                 .contains("Prefer a verified repository tag URL instead.")
-                .contains("If an existing URL can be rendered as a version template, verify the rendered exact-version candidate before writing the \"$version$\" template.");
+                .contains("If an existing URL can be rendered as a version template, verify the rendered candidate for every entry tested-version alias before writing the \"$version$\" template.");
     }
 
     @Test
-    void runUsesEntryMetadataVersionForUrlTemplatesWhenCoordinateMatchesTestedVersion() throws IOException, InterruptedException {
+    void runUsesEntryTestedVersionsForUrlTemplatesWhenCoordinateMatchesTestedVersion() throws IOException, InterruptedException {
         Project project = createProjectWithSharedMetadataVersion();
         PopulateArtifactURLs task = project.getTasks().create(
                 "populateArtifactURLs",
@@ -208,12 +208,13 @@ class PopulateArtifactURLsTests {
         String output = outputBuffer.toString(StandardCharsets.UTF_8);
         assertThat(output)
                 .contains("Find the repository URL, the sources URL, the test suite URL, the documentation URL, and a concise two-sentence explanation for the following library: com.example:demo:1.4.2")
-                .contains("The sources URL, the test suite URL, and the documentation URL must render to the entry metadata-version \"1.4.1\" when \"$version$\" is replaced with \"1.4.1\".")
-                .contains("Set missing \"source-code-url\" to the selected source URL with \"$version$\" replacing version \"1.4.1\".")
-                .contains("Render template candidates for entry metadata-version \"1.4.1\" and verify rendered URLs before writing them.")
+                .contains("The sources URL, the test suite URL, and the documentation URL must render correctly for every entry tested-version alias when \"$version$\" is replaced with that alias. Entry tested-versions: [1.4.2].")
+                .contains("Set missing \"source-code-url\" to the selected source URL with \"$version$\" replacing each tested-version alias only when that template is valid for every alias.")
+                .contains("Render template candidates for every entry tested-version alias and verify rendered URLs before writing them.")
                 .contains("Entry selector: \"metadata-version\" = \"1.4.1\"")
                 .contains("- Coordinate version: 1.4.2")
-                .contains("- Entry metadata-version: 1.4.1");
+                .contains("- Entry metadata-version: 1.4.1")
+                .contains("- Entry tested-versions: [1.4.2]");
     }
 
     private Project createProjectWithMetadata() throws IOException {

--- a/tests/tck-build-logic/src/test/java/org/graalvm/internal/tck/harness/tasks/ValidateIndexFilesTaskTests.java
+++ b/tests/tck-build-logic/src/test/java/org/graalvm/internal/tck/harness/tasks/ValidateIndexFilesTaskTests.java
@@ -1,0 +1,81 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org.graalvm.internal.tck.harness.tasks;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class ValidateIndexFilesTaskTests {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    @Test
+    void detectsMavenCentralUrlTemplatesThatRenderAliasesToUnlistedArtifactVersions() throws IOException {
+        JsonNode index = OBJECT_MAPPER.readTree("""
+                [
+                  {
+                    "metadata-version": "13-ea",
+                    "source-code-url": "https://repo1.maven.org/maven2/org/openjfx/javafx-base/$version$+1/javafx-base-$version$+1-sources.jar",
+                    "documentation-url": "https://repo1.maven.org/maven2/org/openjfx/javafx-base/$version$+1/javafx-base-$version$+1-javadoc.jar",
+                    "tested-versions": [
+                      "13-ea",
+                      "13-ea+1",
+                      "13"
+                    ]
+                  }
+                ]
+                """);
+        List<String> failures = new ArrayList<>();
+
+        ValidateIndexFilesTask.checkLibraryIndexUrlTemplates(
+                index,
+                "metadata/org.openjfx/javafx-base/index.json",
+                failures
+        );
+
+        assertThat(failures)
+                .anySatisfy(failure -> assertThat(failure)
+                        .contains("source-code-url")
+                        .contains("renders tested-version 13-ea+1 to Maven artifact version 13-ea+1+1"))
+                .anySatisfy(failure -> assertThat(failure)
+                        .contains("documentation-url")
+                        .contains("renders tested-version 13 to Maven artifact version 13+1"));
+    }
+
+    @Test
+    void allowsMavenCentralUrlTemplatesThatRenderEachAliasToItsOwnArtifactVersion() throws IOException {
+        JsonNode index = OBJECT_MAPPER.readTree("""
+                [
+                  {
+                    "metadata-version": "1.0.0",
+                    "source-code-url": "https://repo.maven.apache.org/maven2/com/example/demo/$version$/demo-$version$-sources.jar",
+                    "documentation-url": "https://repo.maven.apache.org/maven2/com/example/demo/$version$/demo-$version$-javadoc.jar",
+                    "tested-versions": [
+                      "1.0.0",
+                      "1.1.0"
+                    ]
+                  }
+                ]
+                """);
+        List<String> failures = new ArrayList<>();
+
+        ValidateIndexFilesTask.checkLibraryIndexUrlTemplates(
+                index,
+                "metadata/com.example/demo/index.json",
+                failures
+        );
+
+        assertThat(failures).isEmpty();
+    }
+}

